substrate/frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -93,6 +93,32 @@
 //! Note that both of the bottom solutions end up being discarded and get their deposit back,
 //! despite one of them being *invalid*.
 //!
+//! ### Signed submission deposits
+//!
+//! Each signed submission must hold a deposit that will be returned to the submitter if their
+//! solution was accepted or not processed. The deposit will be slashed if the submission is not
+//! valid. The amount to deposit per submission *may* depend on multiple factors, namely the size of
+//! the submission (in bytes and weight), the deposit base, the inclusion or not of the submitter in
+//! a whitelist and, potentially, on the size of the submission queue.
+//!
+//! The base deposit of a submission is calculated based on whether a submitter is whitelisted. Up
+//! to [`pallet::Config::SignedWhitelistMax`] can be whitelisted at a given time. Everytime a
+//! submitter's solution is verified, valid and thus accepted, the submitter is added to the
+//! whitelist. The whitelist is a bounded FIFO with the list of the last submitters (note: there may
+//! be duplicate submitters in the whitelist, in case their submissions were accepted in the last
```

Comment:
@Ank4n after some discussions, we decided to change the FIFO logic so that there are no duplicate entries and the submitters can now remain for a long time in the whitelist -- similar to what you suggestions. So now the whitelist is updated in a FIFO manner so that:

- Up to `SignedWhitelistMax` of the last successful submitters are part of the whitelist.
- If the new account to add to the whitelist already exists, move it to the front of the queue without touching the order of the remaining accounts.
- If a non-whitelisted account is added to a saturated list, the last (oldest) submitter will be removed.

So no more duplicates.